lib/services/auth.dart

```diff
@@ -0,0 +1,90 @@
+// ignore_for_file: slash_for_doc_comments
+
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter/foundation.dart';
+import 'package:google_sign_in/google_sign_in.dart';
+import 'package:lost_found_steelhacks/services/firebase.dart';
+import 'package:lost_found_steelhacks/models/app_user.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
+
+/**
+ * Singleton Authentication service managing current user state
+ */
+class AuthService extends ChangeNotifier {
+  
+  /* Singleton setup */
+  static final AuthService instance = AuthService._internal();
+  factory AuthService() => instance;
+  AuthService._internal();
+  AppUser? user;
+
+  final GeoPoint defaultGeopoint = const GeoPoint(40.4440279, -79.9700647);
+
+  bool isSignedIn() => FirebaseAuth.instance.currentUser != null;
+
+  Future<void> updateAppUser(String uid) async => user = await FirebaseService.instance.getUser(uid);
+
+  Stream<User?> get authUser => FirebaseAuth.instance.authStateChanges();
+
+  Future registerWithEmailAndPassword(
+      String firstName, String lastName, String email, String password) async {
+    try {
+      UserCredential result = await FirebaseAuth.instance
+          .createUserWithEmailAndPassword(email: email, password: password);
+      User authUser = result.user!;
+      AppUser user = AppUser(
+          uid: authUser.uid,
+          email: authUser.email!,
+          firstName: firstName,
+          lastName: lastName,
+          location: defaultGeopoint);
+      this.user = user;
+      await FirebaseService.instance.createUser(user);
```

Comment:
What is going on with line 41-43?

Reply:
line 42 creates a new user document in Firestore. line 41 appears to be a mistake since this is in the register method not the sign in method. But there's a similar line to 41 in ```signInWithEmailAndPassword``` that keeps the ```AppUser```  instance of ```AuthService``` consistent with the sign in state held in ```FirebaseAuth```. Is there a way we can just get this to automatically update by listening to the FirebaseAuth stream? This would be a better way

```Stream<User?> get authUser => FirebaseAuth.instance.authStateChanges();```
^ this one in AuthService